starboard/build/BUILD.gn

```diff
@@ -0,0 +1,37 @@
+# Copyright 2024 The Cobalt Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import("//build/buildflag_header.gni")
+import("//cobalt/build/configs/cobalt.gni")
+import("//cobalt/build/configs/variables.gni")
+
+declare_args() {
+  use_starboard = target_os != "android"
+  use_evergreen = false
+}
+
+# To use these build flags from i.e. "foo.cc", add a dependency from the target
+# containing "foo.cc" on this target, then have "foo.cc" include
+# "starboard/build/starboard_buildflags.h". After that you can use i.e.
+# BUILDFLAG(USE_STARBOARD) in your code.
+buildflag_header("starboard_buildflags") {
```

Comment:
nice! this is a pretty good improvement from what we were doing with the define  flags